core/BUILD.bazel

```diff
@@ -11,3 +11,28 @@ cc_library(
     strip_include_prefix = "include",
     visibility = ["//visibility:public"],
 )
+
+cc_library(
+    name = "core-headers",
+    hdrs = glob(["include/**/*.h"]),
+    strip_include_prefix = "include",
+)
+
+cc_binary(
```

Comment:
Could you please tell me what's the reason behind separate `cc_binary` and `cc_library` targets?

Reply:
My understanding is that the `cc_binary` target here only provides the shared object file. To use it in some other target, you'd put something like `//core:libcore.so` in the `srcs` field.

If you wrap `libcore.so` in a `cc_library`, you can both expose the headers and use something like `//core:core-shared` in the `deps` field instead. 

Personally, I find it a little awkward to have external targets in my `srcs` fields, but maybe that's not a popular opinion, or I could also have misunderstood how to use such targets.